SampleApplications/2023/ServerlessDocumentAnalysis/functions/test/DocProcessingTest/TestTextractQuery.cs

```diff
@@ -0,0 +1,64 @@
+using DocProcessing.Shared.Model.Textract.QueryAnalysis;
+using System.Text.Json;
+
+
+namespace DocProcessingTest;
+
+[TestClass]
+[DeploymentItem(@"TestAssets\TextractResults.json")]
+public class TestTextractQuery
+{
+    private TextractDataModel? TextractData { get; set; }
+    private TextractAnalysisResult? TextractResult { get; set; }
+
+    [TestInitialize()]
+    public void Setup()
+    {
+        using FileStream jsonStream = File.OpenRead(@"TextractResults.json");
+        if (jsonStream is null)
+        {
+            throw new ArgumentNullException(nameof(jsonStream));
+        }
+        TextractResult = JsonSerializer.Deserialize<TextractAnalysisResult>(jsonStream);
+        if (TextractResult is null)
+        {
+            throw new ArgumentNullException(nameof(TextractResult));
+        }
+        TextractData = new TextractDataModel(TextractResult.Blocks);
+    }
+
+
+    [TestMethod("Test Block Count")]
+    public void TestBlockCount()
+    {
+        Assert.IsTrue(TextractResult?.GetBlockCount() == 1000);
+
+    }
+
+    [TestMethod("Get Query Result")]
+    public void TestQueryResults()
+    {
+        var queryResultPatientName = TextractData?.GetQueryResults("patientname");
+
+        Assert.IsTrue(queryResultPatientName?.Count() == 2);
+
+        Assert.AreEqual(queryResultPatientName.Where(a => a.Text == "Edward Sang").Count(), 1);
```

Comment:
Consider using multiple asserts where applicable (i.e. when testing the same result, different aspects): https://www.code4it.dev/cleancodetips/coherent-levels-of-abstraction/

Reply:
Understood. I simplified it and made what I was testing clearer. I didn't want to overcomplicate this one, for sure.